app/src/main/java/com/babylon/wallet/android/domain/usecases/p2plink/ParseLinkConnectionDetailsUseCase.kt

```diff
@@ -23,49 +23,66 @@ class ParseLinkConnectionDetailsUseCase @Inject constructor(
-                )
-            }.onFailure { throwable ->
-                Timber.e("Failed to verify the link signature: ${content.signature} Error: ${throwable.message}")
-            }.getOrNull() ?: false
+        val newLinkPurpose = P2PLinkPurpose.fromValue(content.purpose)
+        val links = p2PLinksRepository.getP2PLinks()
+        val existingLink = links.findBy(content.publicKey)
 
-            if (!isSignatureValid) {
-                return Result.failure(RadixWalletException.LinkConnectionException.InvalidSignature)
+        return when {
+            newLinkPurpose == null -> {
+                Result.failure(RadixWalletException.LinkConnectionException.UnknownPurpose)
             }
-
-            val newLinkPurpose = P2PLinkPurpose.fromValue(content.purpose)
-            val links = p2PLinksRepository.getP2PLinks()
-            val existingLink = links.findBy(content.publicKey)
-
-            return when {
-                newLinkPurpose == null -> {
-                    Result.failure(RadixWalletException.LinkConnectionException.UnknownPurpose)
-                }
-                existingLink != null && newLinkPurpose != existingLink.purpose -> {
-                    Result.failure(RadixWalletException.LinkConnectionException.PurposeChangeNotSupported)
-                }
-                else -> {
-                    Result.success(
-                        LinkConnectionPayload(
-                            password = RadixConnectPassword(passwordBytes),
-                            publicKey = PublicKey.Ed25519.init(content.publicKey),
-                            purpose = newLinkPurpose,
-                            existingP2PLink = existingLink
-                        )
+            existingLink != null && newLinkPurpose != existingLink.purpose -> {
+                Result.failure(RadixWalletException.LinkConnectionException.PurposeChangeNotSupported)
+            }
+            else -> {
+                Result.success(
+                    LinkConnectionPayload(
+                        password = RadixConnectPassword(Exactly32Bytes.init(content.password.hexToBagOfBytes())),
+                        publicKey = PublicKey.Ed25519.init(content.publicKey),
+                        purpose = newLinkPurpose,
+                        existingP2PLink = existingLink
                     )
-                }
+                )
             }
-        }.onFailure { throwable ->
-            Timber.e("Failed to parse the p2p link connection QR content: $raw Error: ${throwable.message}")
-            return Result.failure(RadixWalletException.LinkConnectionException.InvalidQR)
-        }.getOrThrow<Nothing>()
+        }
+    }
+
+    @Throws(RadixWalletException.LinkConnectionException::class)
+    private fun decodeRawInput(value: String): LinkConnectionQRContent {
+        runCatching {
+            Exactly32Bytes.init(value.hexToBagOfBytes())
+        }.onSuccess {
```

Comment:
not sure I understand why we throw on success here, could you briefly explain what this function does?

Reply:
it checks if the scanned QR is a p2p connection password which means the user scanned an old QR code and has to update the connector extension